### ISSUE TITLE
[$1000] Web – Announce Room - Inconsistency in highlighting the selected option.

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#22553]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Click on "Settings."
2. Click on "Preferences."
3. Click on "Priority mode" and "Language."
4. Notice that the selected option in "Priority mode" and "Language" is highlighted.
5. Open any "announce room."
6. Open "Details."
7. Click on "Settings."
8. Click on "Who can post" and "Notify me."
9. Notice that the selected option is not highlighted.

## Expected Result:
The selected option in "Who can post" and "Notify me" should be highlighted to maintain consistency among all pages.

## Actual Result:
The selected option is not highlighted.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.38-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/f1166e96-bb1e-44af-b5a8-2bfd8730979e

https://github.com/Expensify/App/assets/93399543/4021d943-78c4-4346-8fd1-83c8547de761

**Expensify/Expensify Issue URL:**
**Issue reported by:** @usmantariq96
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688967984457429

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01422c202471bfda96</li>
        <li>Upwork Job ID: 1681657941126709248</li>
        <li>Last Price Increase: 2023-07-19</li>
    </ul>
</details>

**Comment by DanutGavrus** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Web – Announce Room - Inconsistency in highlighting the selected option.

### What is the root cause of that problem?
1. For `Settings` -> `Preferences` -> `Priority mode` & `Language` we are using `SelectionListRadio` which has `initiallyFocusedOptionKey` *property*;
2. In `Announce Rooms` -> `Details` -> `Settings` -> `Who can post` & `Notify me` we are using `OptionsList` which does not have the *initially focused functionality*.

### What changes do you think we should make in order to solve the problem?
We should either:
**I.** Replace the `OptionsList` with `SelectionListRadio`, but keep in mind that `SelectionListRadio` does not have some functionalities which we'll lose such as `shouldHaveOptionSeparator` which adds a green line between rows.
**II.** Implement the *initially focused functionality* for the `OptionsList` same as it works for the `SelectionListRadio`.

### What alternative solutions did you explore?
Didn't investigate enough, but I'm curious if we would be able to use just one list implementation everywhere, and set functionalities based on props. For example, `OptionsList` at `/settings/preferences/language` and `SelectionListRadio` at `/r/YOUR-WORKSPACE-ID/settings/notification-preferences` look very similar. I think this would be a bigger refactor if possible.

**Comment by shawnborton** [comment]
Yeah, @thiagobrez was working on this and can provide some PR links.

**Comment by thiagobrez** [comment]
Hey! 

The selection list refactor tracker issue is here: https://github.com/Expensify/App/issues/11795

Phase 1 - single selection lists:
- Issue: https://github.com/Expensify/App/issues/20352
- PR: https://github.com/Expensify/App/pull/21048

Phase 2 - multi selection lists:
- Issue: https://github.com/Expensify/App/issues/20353
- PR: https://github.com/Expensify/App/pull/22622

Phase 3 - simple selection lists:
- Issue: https://github.com/Expensify/App/issues/20354
- PR: not started yet

---

So what happened seems to be that we forgot about some list cases. I was following the cases listed in [this comment](https://github.com/Expensify/App/issues/11795#issuecomment-1543877225) from the tracker issue, so that's why the Announce room was forgotten.

Before this refactor, we had this component hierarchy:
- `OptionsSelector` parent of `OptionsList` parent of `OptionRow`

These 3 component were being used interchangeably. For example, sometimes `OptionsList` being rendered on it's own without `OptionsSelector` (which is it's parent). That, in addition to a bunch of styling props, made all lists just a little bit different from each other.

We're aiming to replace all usages of `OptionsSelector`, `OptionsList` and `OptionRow` with the new `SelectionList`.

In `main`, you will still find `SelectionListRadio`, which is the result of the Phase 1 PR. In Phase 2 PR, I renamed it to `SelectionList` and iterated on top of it making it also able to render multiple selection lists (with checkboxes instead of checkmarks). On Phase 3, I'm thinking I'll come up with a new component instead of iterating on top of this same one again because the requirements seem quite different, let's see.

TL;DR:
- We want to replace all usages of `OptionsSelector`, `OptionsList` and `OptionRow` with the new `SelectionList`
- Lists that are not using this yet is because we simply forgot to track 🤷🏻 , or by some decision made
- Yes, some styling props (such as `shouldHaveOptionSeparator`) don't exist anymore, and that's intended, to standardise the look and feel of same-purpose lists

**Comment by thiagobrez** [comment]
That said, I believe both Proposals from @DanutGavrus (point I) and @GItGudRatio are correct. You can either raise a PR in this issue, or I can add those to my Phase 2 PR which is currently open, however you prefer.

cc @shawnborton ^

**Comment by DanutGavrus** [comment]
@thiagobrez As my Proposal's variant I and @GItGudRatio's Proposal are the same, shouldn't I be assigned here?

**Comment by JmillsExpensify** [comment]
Let's triage the bug first and C+ can make that call.

**Comment by Santhosh-Sellavel** [comment]
Sorry bit late here I agree with @thiagobrez, we have ongoing migrations so we would handle it there, let's close this one thanks!

cc: @JmillsExpensify @rushatgabhane

**Comment by DanutGavrus** [comment]
@Santhosh-Sellavel @JmillsExpensify @rushatgabhane If this issue was *External* and if my Proposal was left *before* the *same* change was applied in the migrations, will I be eligible for the bounty?  
I think I saw something similar in another Issue recently, but can't find it easily so I thought of asking. Thanks!

**Comment by Santhosh-Sellavel** [comment]
> @Santhosh-Sellavel @JmillsExpensify @rushatgabhane If this issue was _External_ and if my Proposal was left _before_ the _same_ change was applied in the migrations, will I be eligible for the bounty? I think I saw something similar in another Issue recently, but can't find it easily so I thought of asking. Thanks!

We are already in the process of migrating to the selection list. So the proposal doesn't qualify for a bounty.

**Comment by JmillsExpensify** [comment]
Thanks @Santhosh-Sellavel! Closing this one out.
